pay-api/src/pay_api/services/payment_calculations.py

```diff
@@ -0,0 +1,225 @@
+            "transactions": build_transaction_rows(items),
+            "is_index_0": first_group
+        }
+
+        if method == PaymentMethod.EFT.value:
+            method_context["amount_owing"] = get_statement_currency_string(statement.get("amount_owing", 0.00))
+            if statement.get("is_interim_statement") and statement_summary:
+                method_context["latest_payment_date"] = statement_summary.get("latestStatementPaymentDate")
+            elif not statement.get("is_interim_statement") and statement_summary:
+                method_context["due_date"] = get_statement_date_string(statement_summary.get("dueDate"))
+
+        if method == PaymentMethod.INTERNAL.value:
+            has_staff_payment = any("routing_slip" not in inv or inv["routing_slip"] is None for inv in items)
+            method_context["is_staff_payment"] = has_staff_payment
+        else:
+            has_staff_payment = False
+
+        summary = calculate_invoice_summaries(items, method, statement)
+        statement_header_text = StatementTitles['DEFAULT'].value
+
+        if method == PaymentMethod.INTERNAL.value and has_staff_payment:
+            statement_header_text = StatementTitles['INTERNAL_STAFF'].value
+        else:
+            statement_header_text = StatementTitles[method].value
+
+        method_context.update({
+            "paid_summary": summary["paid"],
+            "due_summary": summary["due"],
+            "totals_summary": summary["total"],
+            "statement_header_text": statement_header_text
+        })
+
+        result[method] = method_context
+        first_group = False
+
+    grouped_invoices = [
+        {"payment_method": method, **context}
+        for method, context in result.items()
+    ]
+
+    return grouped_invoices
+
+
+def calculate_invoice_summaries(invoices: List[dict], payment_method: str, statement: dict) -> dict:
+    """Calculate paid, due, and totals summary for a specific payment method."""
+    paid_total, due_total, total_total = 0, 0, 0
+    for invoice in invoices:
+        if invoice.get("payment_method") != payment_method:
+            continue
+
+        paid = Decimal(invoice.get("paid", 0))
+        refund = Decimal(invoice.get("refund", 0))
+        total = Decimal(invoice.get("total", 0))
+        refund_date = invoice.get("refund_date")
+        paid_total += paid
+        total_total += total
+        due_amount = total
+
+        if payment_method != PaymentMethod.EFT.value:
+            due_amount -= paid
+            if paid == 0 and refund > 0:
+                due_amount -= refund
+        else:
+            due_amount -= paid
+            if (
+                paid == 0
+                and refund > 0
+                and refund_date
+                and parser.parse(refund_date) <= parser.parse(statement.get("to_date"))
+            ):
+                due_amount -= refund
+        due_total += due_amount
```

Comment:
wouldn't it make sense to move these to DAO?

So we could easily just re-run a generic query instead of having to run python code?